### PR TITLE
Inital version

### PR DESCRIPTION
**Pull request by macmacal**
## Description
* This is the initial version of the meta-package project for the Aegis UR5e cobot station.
* Based on & inspired by [Husarion's Panther ROS](https://github.com/husarion/panther_ros) and [Universal Robot's ROS2 Tutorial](https://github.com/UniversalRobots/Universal_Robots_ROS2_Tutorials) repositories.
* The PR includes:
  - Four ROS 2 packages (`aegis`, `aegis_bringup`, `aegis_description`, `aegis_moveit_config`).
  - Pre-commit configuration,
  - First GitHub Workflow to run the pre-commit,
  - Issue & PR templates,
  - Minimal documentation to run the project.

> [!IMPORTANT]  
> For this particular PR please perform the `Squash and merge` operation.

### Related PRs:
- https://github.com/AGH-CEAI/aegis_docker/pull/1
- https://github.com/AGH-CEAI/ros2_net_ft_driver/pull/4

## How has this been tested?

- Manually, running the whole project on the geonosis PC (dedicated workstation for the Aegis robot).

## Checklist

- [x] All TODOs in the code has been resolved or linked with a proper issue.
- [x] Code is (auto)formatted.
- [x] Documentation (README/CHANGELOG/Wiki/etc) has been updated.
- [x] All automatic checks passed.

---
Clickup task: [869619jxw](https://app.clickup.com/t/869619jxw)

### CODE REVIEW

File: aegis/aegis.repos [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    type: git
+    url: https://github.com/AGH-CEAI/ros2_net_ft_driver
+    version: 9633eef704d0c3940f4abce9269d842eaf0c9138
+  robotiq_hande_description:
+    type: git
+    url: https://github.com/AGH-CEAI/robotiq_hande_description
+    version: 27a113275f52913b41b27594842ec68fddf7e592
+  scene_objects_manager:
+    type: git
+    url: https://github.com/AGH-CEAI/scene_objects_manager.git
```
Comment (macmacal) [inline aegis/aegis.repos]:
```suggestion
    url: https://github.com/AGH-CEAI/scene_objects_manager
```

File: README.md [hunk 2]
```diff
@@ -1,4 +1,69 @@
...
+#TODO add more examples
+```
+
+#### Podman & Toolbx
+```bash
+toolbox create --image localhost/ceai/aegis_dev
+toolbox enter aegis_dev
+```
+
+### Build
```
Comment (macmacal) [inline README.md]: Add section of cloning and installing `vcs` and `rosdeb` depends

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+<!--- see how your change affects other areas of the code, etc. -->
+
+
+## Checklist
+<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
+<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
+- [ ] All TODOs in the code has been resolved or linked with a proper issue.
+- [ ] Code is (auto)formatted.
+- [ ] Documentation (README/CHANGELOG/Wiki/etc) has been updated.
+- [ ] All automatic checks passed.
```
Comment (macmacal) [inline .github/pull_request_template.md]:
```suggestion
- [ ] All automatic checks passed.

---
Clickup task: [hash](link)
```

File: aegis/CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
+# Changelog
+
+All notable changes to the `aegis` package will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
+
+### Added
```
Comment (macmacal) [inline aegis/CHANGELOG.md:10]: Update changelog

File: aegis/package.xml [hunk 5]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
+  <name>aegis</name>
+  <version>0.0.1</version>
```
Comment (macmacal) [inline aegis/package.xml]: Set the initial version of the all packages to the `0.1.0` to mark out our previous development.

```suggestion
  <version>0.1.0</version>
```

File: aegis_bringup/CHANGELOG.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
+# Changelog
+
+All notable changes to the `aegis_bringup` package will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
+
+### Added
```
Comment (macmacal) [inline aegis_bringup/CHANGELOG.md:10]: Update changelog

File: aegis_bringup/LICENSE
Comment (macmacal) [inline aegis_bringup/LICENSE:1]: We don't need the 2nd LICENSE file

File: aegis_bringup/package.xml [hunk 7]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
+  <name>aegis_bringup</name>
+  <version>0.0.1</version>
```
Comment (macmacal) [inline aegis_bringup/package.xml]:
```suggestion
  <version>0.1.0</version>
```

File: aegis_description/CHANGELOG.md [hunk 8]
```diff
@@ -0,0 +1,20 @@
+# Changelog
+
+All notable changes to the `aegis_description` package will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
+
+### Added
```
Comment (macmacal) [inline aegis_description/CHANGELOG.md:10]: Update changelog

File: aegis_description/README.md [hunk 9]
```diff
@@ -0,0 +1,14 @@
+# aegis_description
+
+This package contains a description of the Aegis robot station, which consists of the following:
```
Comment (macmacal) [inline aegis_description/README.md]:
```suggestion
This package contains a description of the Aegis robot station, which consists of the following modules:
```

File: aegis_description/package.xml [hunk 10]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
+  <name>aegis_description</name>
+  <version>0.0.1</version>
```
Comment (macmacal) [inline aegis_description/package.xml]:
```suggestion
  <version>0.1.0</version>
```

File: aegis_moveit_config/CHANGELOG.md [hunk 11]
```diff
@@ -0,0 +1,20 @@
+# Changelog
+
+All notable changes to the `aegis_moveit_config` package will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
+
+### Added
```
Comment (macmacal) [inline aegis_moveit_config/CHANGELOG.md:10]: Update changelog

File: aegis_moveit_config/package.xml [hunk 12]
```diff
@@ -0,0 +1,53 @@
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
+  <name>aegis_moveit_config</name>
+  <version>0.0.1</version>
```
Comment (macmacal) [inline aegis_moveit_config/package.xml]:
```suggestion
  <version>0.1.0</version>
```

File: .github/pull_request_template.md (hunk 3 above)
Comment (sivral) [inline .github/pull_request_template.md]:
```suggestion
- [ ] All TODOs in the code have been resolved or linked to a proper issue.
- [ ] Code has been (auto)formatted.
- [ ] Documentation (e.g., README, CHANGELOG, Wiki) has been updated.
- [ ] All automated checks have passed.
```

File: aegis/package.xml [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+  <description>Meta package which contains all packages of the Aegis UR station.</description>
+  <maintainer email="ceai@agh.edu.pl">AGH Center of Excelence in Artificial Intelligence</maintainer>
```
Comment (sivral) [inline aegis/package.xml]:
```suggestion
  <maintainer email="ceai@agh.edu.pl">AGH Center of Excellence in Artificial Intelligence</maintainer>
```

File: aegis_bringup/package.xml [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+  <description>Collection of launch files and configuration for running the Aegis robot station</description>
+  <maintainer email="ceai@agh.edu.pl">AGH Center of Excelence in Artificial Intelligence</maintainer>
```
Comment (sivral) [inline aegis_bringup/package.xml]:
```suggestion
  <maintainer email="ceai@agh.edu.pl">AGH Center of Excellence in Artificial Intelligence</maintainer>
```

File: aegis_description/package.xml [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+  <description>Package with description of the Aegis robot station</description>
+  <maintainer email="ceai@agh.edu.pl">AGH Center of Excelence in Artificial Intelligence</maintainer>
```
Comment (sivral) [inline aegis_description/package.xml]:
```suggestion
  <maintainer email="ceai@agh.edu.pl">AGH Center of Excellence in Artificial Intelligence</maintainer>
```

File: aegis_moveit_config/package.xml [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+  <description>An automatically generated package with all the configuration and launch files for using the aegis with the MoveIt Motion Planning Framework</description>
+  <maintainer email="ceai@agh.edu.pl">AGH Center of Excelence in Artificial Intelligence</maintainer>
```
Comment (sivral) [inline aegis_moveit_config/package.xml]:
```suggestion
  <maintainer email="ceai@agh.edu.pl">AGH Center of Excellence in Artificial Intelligence</maintainer>
```

File: aegis/package.xml [hunk 17]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
+  <name>aegis</name>
+  <version>0.0.1</version>
+  <description>Meta package which contains all packages of the Aegis UR station.</description>
```
Comment (sivral) [inline aegis/package.xml]:
```suggestion
  <description>A metapackage for the Aegis UR station</description>
```

File: aegis_bringup/README.md [hunk 18]
```diff
@@ -0,0 +1,38 @@
+# aegis_bringup
+
+The package contains the configuration and launch files necessary to start all the basic functionalities of the Aegis robot station..
```
Comment (sivral) [inline aegis_bringup/README.md]:
```suggestion
This package provides the configuration and launch files required to enable all basic functionalities of the Aegis robot station.
```

File: aegis_bringup/README.md [hunk 19]
```diff
@@ -0,0 +1,38 @@
...
+
+## Launch files
+
+- `bringup.launch.py`: Responsible for running the whole ROS 2 stack for the whole robot station.
```
Comment (sivral) [inline aegis_bringup/README.md]:
```suggestion
- `bringup.launch.py`: Responsible for running the entire ROS 2 stack for the robot station.
```

File: README.md [hunk 20]
```diff
@@ -1,4 +1,69 @@
...
 
+[![pre-commit](https://img.shields.io/badge/pre--commit-enabled-brightgreen?logo=pre-commit)](https://github.com/pre-commit/pre-commit)
+
+---
+
+## List of packages
+
+* [aegis_bringup](./aegis_bringup/README.md) - the main launch file with all hardware-dependend configuration files.
+* [aegis_description](./aegis_description/README.md) - the description of the Aegis robot station.
+* [aegis_moveit_config](./aegis_moveit_config/README.md) - the collection of configuration files to run the [MoveIt2](https://moveit.picknik.ai/main/index.html).
```
Comment (sivral) [inline README.md]:
```suggestion
* [aegis_bringup](./aegis_bringup/README.md): The main launch file with all hardware-dependent configuration files.
* [aegis_description](./aegis_description/README.md): The description of the Aegis robot station.
* [aegis_moveit_config](./aegis_moveit_config/README.md): The collection of configuration files to run the [MoveIt 2](https://moveit.picknik.ai/main/index.html).
```

File: aegis_moveit_config/README.md [hunk 21]
```diff
@@ -0,0 +1,54 @@
...
+```bash
+# Set the robot IP alias in the /etc/hosts file
+ros2 launch ur_robot_driver ur_control.launch.py ur_type:=ur5e robot_ip:=aegis launch_rviz:=false
+```
+2. Ensure that the `ros2_control` is connected the to the robot (play the program via Dashboard):
+```bash
+ros2 service call /dashboard_client/stop std_srvs/srv/Trigger {}
+ros2 service call /dashboard_client/play std_srvs/srv/Trigger {}
+```
+3. Run the MoveIt2
```
Comment (sivral) [inline aegis_moveit_config/README.md]:
```suggestion
3. Launch MoveIt 2 configuration for motion planning with RViz visualization:
```

File: aegis_moveit_config/README.md [hunk 22]
```diff
@@ -0,0 +1,54 @@
...
+```bash
+ros2 launch aegis_moveit_config aegis.launch.py launch_rviz:=true
+```
+
+4. Run the hotfix publisher for the gripper state
```
Comment (sivral) [inline aegis_moveit_config/README.md]:
```suggestion
4. Run the hotfix publisher for the gripper state:
```

File: aegis_moveit_config/package.xml [hunk 23]
```diff
@@ -0,0 +1,53 @@
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
+  <name>aegis_moveit_config</name>
+  <version>0.0.1</version>
+  <description>An automatically generated package with all the configuration and launch files for using the aegis with the MoveIt Motion Planning Framework</description>
```
Comment (sivral) [inline aegis_moveit_config/package.xml]:
```suggestion
  <description>An automatically generated package containing all the configuration and launch files for using the Aegis with the MoveIt Motion Planning Framework</description>
```

File: aegis_moveit_config/urdf/initial_positions.yaml [hunk 24]
```diff
@@ -0,0 +1,10 @@
+# Default initial positions for aegis's ros2_control fake system
```
Comment (sivral) [inline aegis_moveit_config/urdf/initial_positions.yaml]:
```suggestion
# Default initial positions for Aegis' ros2_control fake system
```

File: aegis_description/package.xml [hunk 25]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
+  <name>aegis_description</name>
+  <version>0.0.1</version>
+  <description>Package with description of the Aegis robot station</description>
```
Comment (sivral) [inline aegis_description/package.xml]:
```suggestion
  <description>A package with the description of the Aegis robot station</description>
```

File: aegis_bringup/package.xml [hunk 26]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
+  <name>aegis_bringup</name>
+  <version>0.0.1</version>
+  <description>Collection of launch files and configuration for running the Aegis robot station</description>
```
Comment (sivral) [inline aegis_bringup/package.xml]:
```suggestion
  <description>A collection of launch files and the configuration for running the Aegis robot station</description>
```

**Review by sivral — APPROVED** [review]

**Review by sivral — APPROVED** [review]
Looks good to me!

**Pull request merged**